### PR TITLE
🚧 WIP: Feature/pdf generation

### PR DESCRIPTION
**Pull request by heikomat**
## What did you change?

I added two scripts:

1. One to install the necessary software to render the gitbook as PDF
2. One to actually render the PDF (as npm-script in the package.json)

I also added a small markdown-file that describes how to use them

## How can others test the changes?

You can run `./pdf_generation_preparation_osx.sh`, and when that is done `npm run generate_pdf`, and see how a `documentation.pdf` is generated

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [ ] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.)
- [x] I've merged the `develop` branch into my branch before finishing this PR.
- [x] I've **not added any other changes** than the ones described above.
- [x] I've mentioned all **PRs, which relate to this one**

### CODE REVIEW

File: render_pdf.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# How to Render the Documentation as pdf
```
Comment (cmg-dev) [inline render_pdf.md]: Typo: 

PDF

File: render_pdf.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+- Be on OS X.
+- Make sure you have node installed.
+- run `./pdf_generation_preparation_osx.sh`
+
+you might need to enter your root-password
+
+## Generate the PDF
+
+To generate the PDF, run `npm run generate_pdf`. This will take a while (about 3
+to 4 minutes), and will create a `documentation.pdf`
```
Comment (cmg-dev) [inline render_pdf.md]: I'm generally concerned with the consistency of out work.

This is here a minor issue, but I ask myself: Why are the headlines inconsistently capitalized.
For me, this is an issue with the overall care, we put into our work.

Please be aware of such things.

File: render_pdf.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# How to Render the Documentation as pdf
+
+## Install required software
+
+For the pdf-generation to work, you need to have `svgexport` and `ebook-convert`
```
Comment (cmg-dev) [inline render_pdf.md]: PDF is an abbreviation: PDF ist the correct spelling.

File: render_pdf.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# How to Render the Documentation as pdf
+
+## Install required software
```
Comment (cmg-dev) [inline render_pdf.md]: wrong capitalization.

File: pdf_generation_preparation_osx.sh [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+## symlink ebook-convert to make it runnable
+sudo ln -s /Applications/calibre.app/Contents/MacOS/ebook-convert /usr/local/bin/ebook-convert
+
+## unmount the calibre.dmg
+hdiutil detach -force $TEMP_DIR
+
+## remove the temporary mount-folder and the downloaded calibre.dmg
+rm -r $TEMP_DIR
+rm calibre.dmg
```
Comment (cmg-dev) [inline pdf_generation_preparation_osx.sh]: we will rework this later.

File: render_pdf.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+## Install Required Software
+
+For the PDF-generation to work, you need to have `svgexport` and `ebook-convert`
+installed. To install them:
+
+- Be on OS X.
+- Make sure you have node installed.
+- run `./pdf_generation_preparation_osx.sh`
+
+you might need to enter your root-password
```
Comment (cmg-dev) [inline render_pdf.md]: you will be asked to enter you password, as you need higher privileges.

**Comment by heikomat** [comment]
I'm improving the PDF-generation of the HTTP-API-PDF. Don't merge yet

**Comment by Paulomart** [comment]
@heikomat Whats the status on this PR?

**Comment by Paulomart** [comment]
@heikomat This PR is open since the **15.01.2018**!

Can you please do one of the following:

- address @cmg-dev's comments
- requests a re-review from @cmg-dev 
- close the PR

**Comment by heikomat** [comment]
Closing this as it's getting stale. #71 was openend to not loose the progress made here.

**Pull request closed without merging**